File: backend/initFiles.py
import os
import json

DATA_FOLDER = "data"

REQUIRED_FILES = {
    "clientes.json": [],
    "cuentas.json": [],
    "transacciones.json": []
}

def initFiles():
    if not os.path.exists(DATA_FOLDER):
        os.makedirs(DATA_FOLDER)
        print("📁 Carpeta 'data' creada.")

    for filename, default in REQUIRED_FILES.items():
        path = os.path.join(DATA_FOLDER, filename)

        if not os.path.exists(path):
            with open(path, "w") as f:
                json.dump(default, f, indent=4)
            print(f"Archivo creado: {filename}")

    print("Archivos verificados correctamente.")

File: backend/src/initFiles.js
const fs = require("fs");
const path = require("path");

const dataPath = path.join(__dirname, "..", "data");

const requiredFiles = ["clientes.json", "cuentas.json", "transacciones.json"];

function initFiles() {
  // Crear carpeta /data si no existe
  if (!fs.existsSync(dataPath)) {
    fs.mkdirSync(dataPath);
    console.log("📁 Carpeta /data creada");
  }

  // Crear archivos vacíos si no existen
  requiredFiles.forEach((file) => {
    const filePath = path.join(dataPath, file);

    if (!fs.existsSync(filePath)) {
      fs.writeFileSync(filePath, "[]"); // Puedes usar {} según tu estructura
      console.log(`📄 Archivo creado: ${file}`);
    }
  });

  console.log("✔ Archivos verificados correctamente.");
}

module.exports = initFiles;
